<script type="text/javascript" src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script>
<script async src="https://badge.dimensions.ai/badge.js" charset="utf-8"></script>

<div class="pub_float">
  <div class="__dimensions_badge_embed__" data-doi="10.1038/s42003-022-03679-0" data-hide-zero-citations="false" data-legend="hover-bottom" data-style="small_circle"></div>
  <div class="altmetric-embed" data-badge-type="donut" data-doi="10.1038/s42003-022-03679-0" data-altmetric-id="133374281"></div>
  <span class="pub-title"><a href=" https://doi.org/10.1038/s42003-022-03679-0" target="_new">The gut microbiome variability of a butterflyfish increases on severely degraded Caribbean reefs</a></span>
  <span class="pub-authors">Clever F<sup>☯</sup>, Sourisse JM<sup>☯</sup>, Preziosi RF, Eisen JA, Guerra ECR, Scott JJ, Wilkins LGE, Altieri AH, McMillan WO, Leray M. </span>
  <span class="pub-co-first-authors"><sup>☯</sup>Co-first authors</span>
</div>


> [Click here](https://www.nature.com/articles/s42003-022-03679-0/metrics) for article metrics on the  Communications Biology website. 

<br/>

This site provides reproducible workflows for the study on the gut microbiome of Caribbean butterflyfish. If you want to follow the workflow step-by-step, please use the navigation bar above or the section below, where scripts are presented in order. If you would instead like to jump to specific analyses, please select an area on the **clickable image map** to go directly to that section.


## Script Details

[No 1. Indicator Analysis](wf1.html): Script to identify the core fish gut microbiome using Indicator Analysis.

[No 2. Comparing Sample Fractions](wf2.html): Script to assess the relative abundance of phylum & genus level diversity across sample fractions.

[No 3. Subset Fish Only Samples](wf3.html): Script to subset the fish gut only microbiome from the rarefied data for subsequent analyses.

[No 4. Alpha Diversity Estimates](wf4.html): Script testing alpha diversity (Hill Diversity, Shannon Exponential, & Simpson Index) against the whole and core fish gut communities.

[No 5. Beta Diversity Estimates](wf5.html): Script testing different distance metrics to estimate beta diversity using the whole and core fish gut communities across a range of environmental variables.

[No 6. Beta Dispersion](wf6.html): Script to generate faceted boxplot of beta dispersion for all six alpha diversity metrics from the whole & core fish gut microbiomes.

[No 7. PIME Analysis](wf7.html): Script to run Prevalence Interval for Microbiome Evaluation (PIME) to identify core fish gut ASVs.



## Credits {.appendix}

The banner photo, entitled **Butterfly-fish, *Chaetodon capistratus* Linnaeus**,  was retrieved from Wikimedia Commons and licensed under [CC-0](https://creativecommons.org/share-your-work/public-domain/cc0). The image is part of the [Freshwater and Marine Image Bank](https://content.lib.washington.edu/fishweb/index.html) at the University of Washington. 

## Acknowledgements {.appendix}

We thank Lucia Rodriguez for field assistance, Joan Antaneda for her help in the laboratory, Ross Whippo for conducting the fish survey and Clare Fieseler for taking photos of the benthos. The staff of the Bocas del Toro Research Station provided logistical support. We are grateful to Kristin Saltonstall and Marta Vargas for their support at the Smithsonian Tropical Research Institute’s (STRI) Ecological and Evolutionary Genomics Laboratory. Friederike Clever was supported by a Smithsonian Short Term Fellowship. This project was funded, in part, by a grant from the Gordon and Betty Moore foundation to STRI and UC Davis (PIs: William Wcislo and Jonathan Eisen; http://doi.org/10.37807/GBMF5603) and a PhD studentship by Manchester Metropolitan University to Friederike Clever. A research permit was issued by the Ministerio de Ambiente Panamá (No. SE/A-113-17).

## Author Contributions {.appendix}

FC, ML, and JJS conceived the study. ML, FC, and RFP designed the study with input from AHA. FC, ML, and JJS conducted the fieldwork. ECRG and FC dissected the fish guts. FC extracted the DNA. JS and ML prepared the DNA for sequencing and processed the sequencing data. RFP, AHA, WOM, and ML contributed reagents and supplies. ECRG analysed the photographic benthic quadrats. FC, JS, and ML analysed the data and wrote the first draft of the manuscript with input from RFP and LGEW. All authors reviewed the manuscript and contributed to the final version. JJS and FC created the website.

## Data Availability {.appendix}

Sequencing data has been submitted to the NCBI Short Read Archive (SRA) database (https://www.ncbi.nlm.nih.gov/sra) under project number XXX. 

## Source Code {.appendix}

The source code for this page can be accessed on GitHub by [clicking this link](https://github.com/bocasbiome/web/blob/master/index.Rmd). Source code for the complete site can be accessed [here](https://github.com/bocasbiome/web/) or by clicking  the icon on the far right of the navigation bar.

