<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">

<head>
  <meta charset="utf-8"/>
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta http-equiv="X-UA-Compatible" content="IE=Edge,chrome=1"/>
  <meta name="generator" content="distill" />

  <style type="text/css">
  /* Hide doc at startup (prevent jankiness while JS renders/transforms) */
  body {
    visibility: hidden;
  }
  </style>

 <!--radix_placeholder_import_source-->
 <!--/radix_placeholder_import_source-->

  <!--radix_placeholder_meta_tags-->
  <title>: No 1. Indicator Analysis</title>

  <meta property="description" itemprop="description" content="Script to identify the core fish gut microbiome using Indicator Analysis."/>

  <link rel="license" href="https://creativecommons.org/licenses/by/4.0/"/>


  <!--  https://developers.facebook.com/docs/sharing/webmasters#markup -->
  <meta property="og:title" content=": No 1. Indicator Analysis"/>
  <meta property="og:type" content="article"/>
  <meta property="og:description" content="Script to identify the core fish gut microbiome using Indicator Analysis."/>
  <meta property="og:locale" content="en_US"/>
  <meta property="og:site_name" content=""/>

  <!--  https://dev.twitter.com/cards/types/summary -->
  <meta property="twitter:card" content="summary"/>
  <meta property="twitter:title" content=": No 1. Indicator Analysis"/>
  <meta property="twitter:description" content="Script to identify the core fish gut microbiome using Indicator Analysis."/>

  <!--/radix_placeholder_meta_tags-->

  <!--radix_placeholder_rmarkdown_metadata-->

  <script type="text/json" id="radix-rmarkdown-metadata">
  {"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["title","description","bibliography"]}},"value":[{"type":"character","attributes":{},"value":["No 1. Indicator Analysis"]},{"type":"character","attributes":{},"value":["Script to identify the core fish gut microbiome using Indicator Analysis.\n"]},{"type":"character","attributes":{},"value":["assets/cite.bib"]}]}
  </script>
  <!--/radix_placeholder_rmarkdown_metadata-->
  <!--radix_placeholder_navigation_in_header-->

  <script type="application/javascript">

    window.headroom_prevent_pin = false;

    window.document.addEventListener("DOMContentLoaded", function (event) {

      // initialize headroom for banner
      var header = $('header').get(0);
      var headerHeight = header.offsetHeight;
      var headroom = new Headroom(header, {
        onPin : function() {
          if (window.headroom_prevent_pin) {
            window.headroom_prevent_pin = false;
            headroom.unpin();
          }
        }
      });
      headroom.init();
      if(window.location.hash)
        headroom.unpin();
      $(header).addClass('headroom--transition');

      // offset scroll location for banner on hash change
      // (see: https://github.com/WickyNilliams/headroom.js/issues/38)
      window.addEventListener("hashchange", function(event) {
        window.scrollTo(0, window.pageYOffset - (headerHeight + 25));
      });

      // responsive menu
      $('.distill-site-header').each(function(i, val) {
        var topnav = $(this);
        var toggle = topnav.find('.nav-toggle');
        toggle.on('click', function() {
          topnav.toggleClass('responsive');
        });
      });

      // nav dropdowns
      $('.nav-dropbtn').click(function(e) {
        $(this).next('.nav-dropdown-content').toggleClass('nav-dropdown-active');
        $(this).parent().siblings('.nav-dropdown')
           .children('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $("body").click(function(e){
        $('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $(".nav-dropdown").click(function(e){
        e.stopPropagation();
      });
    });
  </script>

  <style type="text/css">

  /* Theme (user-documented overrideables for nav appearance) */

  .distill-site-nav {
    color: rgba(255, 255, 255, 0.8);
    background-color: #455a64;
    font-size: 15px;
    font-weight: 300;
  }

  .distill-site-nav a {
    color: inherit;
    text-decoration: none;
  }

  .distill-site-nav a:hover {
    color: white;
  }

  @media print {
    .distill-site-nav {
      display: none;
    }
  }

  .distill-site-header {

  }

  .distill-site-footer {

  }


  /* Site Header */

  .distill-site-header {
    width: 100%;
    box-sizing: border-box;
    z-index: 3;
  }

  .distill-site-header .nav-left {
    display: inline-block;
    margin-left: 8px;
  }

  @media screen and (max-width: 768px) {
    .distill-site-header .nav-left {
      margin-left: 0;
    }
  }


  .distill-site-header .nav-right {
    float: right;
    margin-right: 8px;
  }

  .distill-site-header a,
  .distill-site-header .title {
    display: inline-block;
    text-align: center;
    padding: 14px 10px 14px 10px;
  }

  .distill-site-header .title {
    font-size: 18px;
  }

  .distill-site-header .logo {
    padding: 0;
  }

  .distill-site-header .logo img {
    display: none;
    max-height: 20px;
    width: auto;
    margin-bottom: -4px;
  }

  .distill-site-header .nav-image img {
    max-height: 18px;
    width: auto;
    display: inline-block;
    margin-bottom: -3px;
  }



  @media screen and (min-width: 1000px) {
    .distill-site-header .logo img {
      display: inline-block;
    }
    .distill-site-header .nav-left {
      margin-left: 20px;
    }
    .distill-site-header .nav-right {
      margin-right: 20px;
    }
    .distill-site-header .title {
      padding-left: 12px;
    }
  }


  .distill-site-header .nav-toggle {
    display: none;
  }

  .nav-dropdown {
    display: inline-block;
    position: relative;
  }

  .nav-dropdown .nav-dropbtn {
    border: none;
    outline: none;
    color: rgba(255, 255, 255, 0.8);
    padding: 16px 10px;
    background-color: transparent;
    font-family: inherit;
    font-size: inherit;
    font-weight: inherit;
    margin: 0;
    margin-top: 1px;
    z-index: 2;
  }

  .nav-dropdown-content {
    display: none;
    position: absolute;
    background-color: white;
    min-width: 200px;
    border: 1px solid rgba(0,0,0,0.15);
    border-radius: 4px;
    box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.1);
    z-index: 1;
    margin-top: 2px;
    white-space: nowrap;
    padding-top: 4px;
    padding-bottom: 4px;
  }

  .nav-dropdown-content hr {
    margin-top: 4px;
    margin-bottom: 4px;
    border: none;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }

  .nav-dropdown-active {
    display: block;
  }

  .nav-dropdown-content a, .nav-dropdown-content .nav-dropdown-header {
    color: black;
    padding: 6px 24px;
    text-decoration: none;
    display: block;
    text-align: left;
  }

  .nav-dropdown-content .nav-dropdown-header {
    display: block;
    padding: 5px 24px;
    padding-bottom: 0;
    text-transform: uppercase;
    font-size: 14px;
    color: #999999;
    white-space: nowrap;
  }

  .nav-dropdown:hover .nav-dropbtn {
    color: white;
  }

  .nav-dropdown-content a:hover {
    background-color: #ddd;
    color: black;
  }

  .nav-right .nav-dropdown-content {
    margin-left: -45%;
    right: 0;
  }

  @media screen and (max-width: 768px) {
    .distill-site-header a, .distill-site-header .nav-dropdown  {display: none;}
    .distill-site-header a.nav-toggle {
      float: right;
      display: block;
    }
    .distill-site-header .title {
      margin-left: 0;
    }
    .distill-site-header .nav-right {
      margin-right: 0;
    }
    .distill-site-header {
      overflow: hidden;
    }
    .nav-right .nav-dropdown-content {
      margin-left: 0;
    }
  }


  @media screen and (max-width: 768px) {
    .distill-site-header.responsive {position: relative;}
    .distill-site-header.responsive a.nav-toggle {
      position: absolute;
      right: 0;
      top: 0;
    }
    .distill-site-header.responsive a,
    .distill-site-header.responsive .nav-dropdown {
      display: block;
      text-align: left;
    }
    .distill-site-header.responsive .nav-left,
    .distill-site-header.responsive .nav-right {
      width: 100%;
    }
    .distill-site-header.responsive .nav-dropdown {float: none;}
    .distill-site-header.responsive .nav-dropdown-content {position: relative;}
    .distill-site-header.responsive .nav-dropdown .nav-dropbtn {
      display: block;
      width: 100%;
      text-align: left;
    }
  }

  /* Site Footer */

  .distill-site-footer {
    width: 100%;
    overflow: hidden;
    box-sizing: border-box;
    z-index: 3;
    margin-top: 30px;
    padding-top: 30px;
    padding-bottom: 30px;
    text-align: center;
  }

  /* Headroom */

  d-title {
    padding-top: 6rem;
  }

  @media print {
    d-title {
      padding-top: 4rem;
    }
  }

  .headroom {
    z-index: 1000;
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
  }

  .headroom--transition {
    transition: all .4s ease-in-out;
  }

  .headroom--unpinned {
    top: -100px;
  }

  .headroom--pinned {
    top: 0;
  }

  </style>

  <link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet"/>
  <link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet"/>
  <script src="site_libs/headroom-0.9.4/headroom.min.js"></script>
  <!--/radix_placeholder_navigation_in_header-->
  <!--radix_placeholder_distill-->

  <style type="text/css">

  body {
    background-color: white;
  }

  .pandoc-table {
    width: 100%;
  }

  .pandoc-table>caption {
    margin-bottom: 10px;
  }

  .pandoc-table th:not([align]) {
    text-align: left;
  }

  .pagedtable-footer {
    font-size: 15px;
  }

  .html-widget {
    margin-bottom: 2.0em;
  }

  .l-screen-inset {
    padding-right: 16px;
  }

  .l-screen .caption {
    margin-left: 10px;
  }

  .shaded {
    background: rgb(247, 247, 247);
    padding-top: 20px;
    padding-bottom: 20px;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }

  .shaded .html-widget {
    margin-bottom: 0;
    border: 1px solid rgba(0, 0, 0, 0.1);
  }

  .shaded .shaded-content {
    background: white;
  }

  .text-output {
    margin-top: 0;
    line-height: 1.5em;
  }

  .hidden {
    display: none !important;
  }

  d-article {
    padding-bottom: 30px;
  }

  d-appendix {
    padding-top: 30px;
  }

  d-article>p>img {
    width: 100%;
  }

  d-article iframe {
    border: 1px solid rgba(0, 0, 0, 0.1);
    margin-bottom: 2.0em;
    width: 100%;
  }

  figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }

  /* CSS for table of contents */

  .d-toc {
    color: rgba(0,0,0,0.8);
    font-size: 0.8em;
    line-height: 1em;
  }

  .d-toc-header {
    font-size: 0.6rem;
    font-weight: 400;
    color: rgba(0, 0, 0, 0.5);
    text-transform: uppercase;
    margin-top: 0;
    margin-bottom: 1.3em;
  }

  .d-toc a {
    border-bottom: none;
  }

  .d-toc ul {
    padding-left: 0;
  }

  .d-toc li>ul {
    padding-top: 0.8em;
    padding-left: 16px;
    margin-bottom: 0.6em;
  }

  .d-toc ul,
  .d-toc li {
    list-style-type: none;
  }

  .d-toc li {
    margin-bottom: 0.9em;
  }

  .d-toc-separator {
    margin-top: 20px;
    margin-bottom: 2em;
  }

  .d-article-with-toc {
    border-top: none;
    padding-top: 0;
  }



  /* Tweak code blocks (note that this CSS is repeated above in an injection
     into the d-code shadow dom) */

  d-code {
    overflow-x: auto !important;
  }

  pre.d-code code.d-code {
    padding-left: 10px;
    font-size: 12px;
    border-left: 2px solid rgba(0,0,0,0.1);
  }

  pre.text-output {

    font-size: 12px;
    color: black;
    background: none;
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;

    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;

    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }

  @media(min-width: 768px) {

  d-code {
    overflow-x: visible !important;
  }

  pre.d-code code.d-code  {
      padding-left: 18px;
      font-size: 14px;
  }
  pre.text-output {
    font-size: 14px;
  }
  }

  /* Figure */

  .figure {
    position: relative;
    margin-bottom: 2.5em;
    margin-top: 1.5em;
  }

  .figure img {
    width: 100%;
  }

  .figure .caption {
    color: rgba(0, 0, 0, 0.6);
    font-size: 12px;
    line-height: 1.5em;
  }

  .figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }

  .figure .caption a {
    color: rgba(0, 0, 0, 0.6);
  }

  .figure .caption b,
  .figure .caption strong, {
    font-weight: 600;
    color: rgba(0, 0, 0, 1.0);
  }



  /* Tweak 1000px media break to show more text */

  @media(min-width: 1000px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 80px [middle-start] 50px [text-start kicker-end] 65px 65px 65px 65px 65px 65px 65px 65px [text-end gutter-start] 65px [middle-end] 65px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 16px;
    }

    .grid {
      grid-column-gap: 16px;
    }

    d-article {
      font-size: 1.06rem;
      line-height: 1.7em;
    }
    figure .caption, .figure .caption, figure figcaption {
      font-size: 13px;
    }
  }

  @media(min-width: 1180px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 60px [middle-start] 60px [text-start kicker-end] 60px 60px 60px 60px 60px 60px 60px 60px [text-end gutter-start] 60px [middle-end] 60px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 32px;
    }

    .grid {
      grid-column-gap: 32px;
    }
  }


  /* Get the citation styles for the appendix (not auto-injected on render since
     we do our own rendering of the citation appendix) */

  d-appendix .citation-appendix,
  .d-appendix .citation-appendix {
    font-size: 11px;
    line-height: 15px;
    border-left: 1px solid rgba(0, 0, 0, 0.1);
    padding-left: 18px;
    border: 1px solid rgba(0,0,0,0.1);
    background: rgba(0, 0, 0, 0.02);
    padding: 10px 18px;
    border-radius: 3px;
    color: rgba(150, 150, 150, 1);
    overflow: hidden;
    margin-top: -12px;
    white-space: pre-wrap;
    word-wrap: break-word;
  }


  /* Social footer */

  .social_footer {
    margin-top: 30px;
    margin-bottom: 0;
    color: rgba(0,0,0,0.67);
  }

  .disqus-comments {
    margin-right: 30px;
  }

  .disqus-comment-count {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
    cursor: pointer;
  }

  #disqus_thread {
    margin-top: 30px;
  }

  .article-sharing a {
    border-bottom: none;
    margin-right: 8px;
  }

  .article-sharing a:hover {
    border-bottom: none;
  }

  .sidebar-section.subscribe {
    font-size: 12px;
    line-height: 1.6em;
  }

  .subscribe p {
    margin-bottom: 0.5em;
  }


  .article-footer .subscribe {
    font-size: 15px;
    margin-top: 45px;
  }


  .sidebar-section.custom {
    font-size: 12px;
    line-height: 1.6em;
  }

  .custom p {
    margin-bottom: 0.5em;
  }


  /* Improve display for browsers without grid (IE/Edge <= 15) */

  .downlevel {
    line-height: 1.6em;
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
    margin: 0;
  }

  .downlevel .d-title {
    padding-top: 6rem;
    padding-bottom: 1.5rem;
  }

  .downlevel .d-title h1 {
    font-size: 50px;
    font-weight: 700;
    line-height: 1.1em;
    margin: 0 0 0.5rem;
  }

  .downlevel .d-title p {
    font-weight: 300;
    font-size: 1.2rem;
    line-height: 1.55em;
    margin-top: 0;
  }

  .downlevel .d-byline {
    padding-top: 0.8em;
    padding-bottom: 0.8em;
    font-size: 0.8rem;
    line-height: 1.8em;
  }

  .downlevel .section-separator {
    border: none;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
  }

  .downlevel .d-article {
    font-size: 1.06rem;
    line-height: 1.7em;
    padding-top: 1rem;
    padding-bottom: 2rem;
  }


  .downlevel .d-appendix {
    padding-left: 0;
    padding-right: 0;
    max-width: none;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.5);
    padding-top: 40px;
    padding-bottom: 48px;
  }

  .downlevel .footnotes ol {
    padding-left: 13px;
  }

  .downlevel .base-grid,
  .downlevel .distill-header,
  .downlevel .d-title,
  .downlevel .d-abstract,
  .downlevel .d-article,
  .downlevel .d-appendix,
  .downlevel .distill-appendix,
  .downlevel .d-byline,
  .downlevel .d-footnote-list,
  .downlevel .d-citation-list,
  .downlevel .distill-footer,
  .downlevel .appendix-bottom,
  .downlevel .posts-container {
    padding-left: 40px;
    padding-right: 40px;
  }

  @media(min-width: 768px) {
    .downlevel .base-grid,
    .downlevel .distill-header,
    .downlevel .d-title,
    .downlevel .d-abstract,
    .downlevel .d-article,
    .downlevel .d-appendix,
    .downlevel .distill-appendix,
    .downlevel .d-byline,
    .downlevel .d-footnote-list,
    .downlevel .d-citation-list,
    .downlevel .distill-footer,
    .downlevel .appendix-bottom,
    .downlevel .posts-container {
    padding-left: 150px;
    padding-right: 150px;
    max-width: 900px;
  }
  }

  .downlevel pre code {
    display: block;
    border-left: 2px solid rgba(0, 0, 0, .1);
    padding: 0 0 0 20px;
    font-size: 14px;
  }

  .downlevel code, .downlevel pre {
    color: black;
    background: none;
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;

    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;

    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }

  </style>

  <script type="application/javascript">

  function is_downlevel_browser() {
    if (bowser.isUnsupportedBrowser({ msie: "12", msedge: "16"},
                                   window.navigator.userAgent)) {
      return true;
    } else {
      return window.load_distill_framework === undefined;
    }
  }

  // show body when load is complete
  function on_load_complete() {

    // set body to visible
    document.body.style.visibility = 'visible';

    // force redraw for leaflet widgets
    if (window.HTMLWidgets) {
      var maps = window.HTMLWidgets.findAll(".leaflet");
      $.each(maps, function(i, el) {
        var map = this.getMap();
        map.invalidateSize();
        map.eachLayer(function(layer) {
          if (layer instanceof L.TileLayer)
            layer.redraw();
        });
      });
    }

    // trigger 'shown' so htmlwidgets resize
    $('d-article').trigger('shown');
  }

  function init_distill() {

    init_common();

    // create front matter
    var front_matter = $('<d-front-matter></d-front-matter>');
    $('#distill-front-matter').wrap(front_matter);

    // create d-title
    $('.d-title').changeElementType('d-title');

    // create d-byline
    var byline = $('<d-byline></d-byline>');
    $('.d-byline').replaceWith(byline);

    // create d-article
    var article = $('<d-article></d-article>');
    $('.d-article').wrap(article).children().unwrap();

    // move posts container into article
    $('.posts-container').appendTo($('d-article'));

    // create d-appendix
    $('.d-appendix').changeElementType('d-appendix');

    // create d-bibliography
    var bibliography = $('<d-bibliography></d-bibliography>');
    $('#distill-bibliography').wrap(bibliography);

    // flag indicating that we have appendix items
    var appendix = $('.appendix-bottom').children('h3').length > 0;

    // replace citations with <d-cite>
    $('.citation').each(function(i, val) {
      appendix = true;
      var cites = $(this).attr('data-cites').split(" ");
      var dt_cite = $('<d-cite></d-cite>');
      dt_cite.attr('key', cites.join());
      $(this).replaceWith(dt_cite);
    });
    // remove refs
    $('#refs').remove();

    // replace footnotes with <d-footnote>
    $('.footnote-ref').each(function(i, val) {
      appendix = true;
      var href = $(this).attr('href');
      var id = href.replace('#', '');
      var fn = $('#' + id);
      var fn_p = $('#' + id + '>p');
      fn_p.find('.footnote-back').remove();
      var text = fn_p.html();
      var dtfn = $('<d-footnote></d-footnote>');
      dtfn.html(text);
      $(this).replaceWith(dtfn);
    });
    // remove footnotes
    $('.footnotes').remove();

    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      var id = $(this).attr('id');
      $('.d-toc a[href="#' + id + '"]').parent().remove();
      appendix = true;
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('d-appendix'));
    });

    // show d-appendix if we have appendix content
    $("d-appendix").css('display', appendix ? 'grid' : 'none');

    // replace code blocks with d-code
    $('pre>code').each(function(i, val) {
      var code = $(this);
      var pre = code.parent();
      var clz = "";
      var language = pre.attr('class');
      if (language) {
        // map unknown languages to "clike" (without this they just dissapear)
        if ($.inArray(language, ["bash", "clike", "css", "go", "html",
                                 "javascript", "js", "julia", "lua", "markdown",
                                 "markup", "mathml", "python", "svg", "xml"]) == -1)
          language = "clike";
        language = ' language="' + language + '"';
        var dt_code = $('<d-code block' + language + clz + '></d-code>');
        dt_code.text(code.text());
        pre.replaceWith(dt_code);
      } else {
        code.addClass('text-output').unwrap().changeElementType('pre');
      }
    });

    // localize layout chunks to just output
    $('.layout-chunk').each(function(i, val) {

      // capture layout
      var layout = $(this).attr('data-layout');

      // apply layout to markdown level block elements
      var elements = $(this).children().not('d-code, pre.text-output, script');
      elements.each(function(i, el) {
        var layout_div = $('<div class="' + layout + '"></div>');
        if (layout_div.hasClass('shaded')) {
          var shaded_content = $('<div class="shaded-content"></div>');
          $(this).wrap(shaded_content);
          $(this).parent().wrap(layout_div);
        } else {
          $(this).wrap(layout_div);
        }
      });


      // unwrap the layout-chunk div
      $(this).children().unwrap();
    });

    // load distill framework
    load_distill_framework();

    // wait for window.distillRunlevel == 4 to do post processing
    function distill_post_process() {

      if (!window.distillRunlevel || window.distillRunlevel < 4)
        return;

      // hide author/affiliations entirely if we have no authors
      var front_matter = JSON.parse($("#distill-front-matter").html());
      var have_authors = front_matter.authors && front_matter.authors.length > 0;
      if (!have_authors)
        $('d-byline').addClass('hidden');

      // table of contents
      if (have_authors) // adjust border if we are in authors
        $('.d-toc').parent().addClass('d-article-with-toc');

      // strip links that point to #
      $('.authors-affiliations').find('a[href="#"]').removeAttr('href');

      // hide elements of author/affiliations grid that have no value
      function hide_byline_column(caption) {
        $('d-byline').find('h3:contains("' + caption + '")').parent().css('visibility', 'hidden');
      }

      // affiliations
      var have_affiliations = false;
      for (var i = 0; i<front_matter.authors.length; ++i) {
        var author = front_matter.authors[i];
        if (author.affiliation !== "&nbsp;") {
          have_affiliations = true;
          break;
        }
      }
      if (!have_affiliations)
        $('d-byline').find('h3:contains("Affiliations")').css('visibility', 'hidden');

      // published date
      if (!front_matter.publishedDate)
        hide_byline_column("Published");

      // document object identifier
      var doi = $('d-byline').find('h3:contains("DOI")');
      var doi_p = doi.next().empty();
      if (!front_matter.doi) {
        // if we have a citation and valid citationText then link to that
        if ($('#citation').length > 0 && front_matter.citationText) {
          doi.html('Citation');
          $('<a href="#citation"></a>')
            .text(front_matter.citationText)
            .appendTo(doi_p);
        } else {
          hide_byline_column("DOI");
        }
      } else {
        $('<a></a>')
           .attr('href', "https://doi.org/" + front_matter.doi)
           .html(front_matter.doi)
           .appendTo(doi_p);
      }

       // change plural form of authors/affiliations
      if (front_matter.authors.length === 1) {
        var grid = $('.authors-affiliations');
        grid.children('h3:contains("Authors")').text('Author');
        grid.children('h3:contains("Affiliations")').text('Affiliation');
      }

      // inject pre code styles (can't do this with a global stylesheet b/c a shadow root is used)
      $('d-code').each(function(i, val) {
        var style = document.createElement('style');
        style.innerHTML = 'pre code { padding-left: 10px; font-size: 12px; border-left: 2px solid rgba(0,0,0,0.1); } ' +
                          '@media(min-width: 768px) { pre code { padding-left: 18px; font-size: 14px; } }';
        if (this.shadowRoot)
          this.shadowRoot.appendChild(style);
      });

      // move appendix-bottom entries to the bottom
      $('.appendix-bottom').appendTo('d-appendix').children().unwrap();
      $('.appendix-bottom').remove();

      // clear polling timer
      clearInterval(tid);

      // show body now that everything is ready
      on_load_complete();
    }

    var tid = setInterval(distill_post_process, 50);
    distill_post_process();

  }

  function init_downlevel() {

    init_common();

     // insert hr after d-title
    $('.d-title').after($('<hr class="section-separator"/>'));

    // check if we have authors
    var front_matter = JSON.parse($("#distill-front-matter").html());
    var have_authors = front_matter.authors && front_matter.authors.length > 0;

    // manage byline/border
    if (!have_authors)
      $('.d-byline').remove();
    $('.d-byline').after($('<hr class="section-separator"/>'));
    $('.d-byline a').remove();

    // remove toc
    $('.d-toc-header').remove();
    $('.d-toc').remove();
    $('.d-toc-separator').remove();

    // move appendix elements
    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('.d-appendix'));
    });


    // inject headers into references and footnotes
    var refs_header = $('<h3></h3>');
    refs_header.text('References');
    $('#refs').prepend(refs_header);

    var footnotes_header = $('<h3></h3');
    footnotes_header.text('Footnotes');
    $('.footnotes').children('hr').first().replaceWith(footnotes_header);

    // move appendix-bottom entries to the bottom
    $('.appendix-bottom').appendTo('.d-appendix').children().unwrap();
    $('.appendix-bottom').remove();

    // remove appendix if it's empty
    if ($('.d-appendix').children().length === 0)
      $('.d-appendix').remove();

    // prepend separator above appendix
    $('.d-appendix').before($('<hr class="section-separator" style="clear: both"/>'));

    // trim code
    $('pre>code').each(function(i, val) {
      $(this).html($.trim($(this).html()));
    });

    // move posts-container right before article
    $('.posts-container').insertBefore($('.d-article'));

    $('body').addClass('downlevel');

    on_load_complete();
  }


  function init_common() {

    // jquery plugin to change element types
    (function($) {
      $.fn.changeElementType = function(newType) {
        var attrs = {};

        $.each(this[0].attributes, function(idx, attr) {
          attrs[attr.nodeName] = attr.nodeValue;
        });

        this.replaceWith(function() {
          return $("<" + newType + "/>", attrs).append($(this).contents());
        });
      };
    })(jQuery);

    // prevent underline for linked images
    $('a > img').parent().css({'border-bottom' : 'none'});

    // mark non-body figures created by knitr chunks as 100% width
    $('.layout-chunk').each(function(i, val) {
      var figures = $(this).find('img, .html-widget');
      if ($(this).attr('data-layout') !== "l-body") {
        figures.css('width', '100%');
      } else {
        figures.css('max-width', '100%');
        figures.filter("[width]").each(function(i, val) {
          var fig = $(this);
          fig.css('width', fig.attr('width') + 'px');
        });

      }
    });

    // auto-append index.html to post-preview links in file: protocol
    // and in rstudio ide preview
    $('.post-preview').each(function(i, val) {
      if (window.location.protocol === "file:")
        $(this).attr('href', $(this).attr('href') + "index.html");
    });

    // get rid of index.html references in header
    if (window.location.protocol !== "file:") {
      $('.distill-site-header a[href]').each(function(i,val) {
        $(this).attr('href', $(this).attr('href').replace("index.html", "./"));
      });
    }

    // add class to pandoc style tables
    $('tr.header').parent('thead').parent('table').addClass('pandoc-table');
    $('.kable-table').children('table').addClass('pandoc-table');

    // add figcaption style to table captions
    $('caption').parent('table').addClass("figcaption");

    // initialize posts list
    if (window.init_posts_list)
      window.init_posts_list();

    // implmement disqus comment link
    $('.disqus-comment-count').click(function() {
      window.headroom_prevent_pin = true;
      $('#disqus_thread').toggleClass('hidden');
      if (!$('#disqus_thread').hasClass('hidden')) {
        var offset = $(this).offset();
        $(window).resize();
        $('html, body').animate({
          scrollTop: offset.top - 35
        });
      }
    });
  }

  document.addEventListener('DOMContentLoaded', function() {
    if (is_downlevel_browser())
      init_downlevel();
    else
      window.addEventListener('WebComponentsReady', init_distill);
  });

  </script>

  <!--/radix_placeholder_distill-->
  <script src="site_libs/htmlwidgets-1.5.1/htmlwidgets.js"></script>
  <script src="site_libs/jquery-1.12.4/jquery.min.js"></script>
  <link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
  <script src="site_libs/datatables-binding-0.14/datatables.js"></script>
  <link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
  <link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
  <script src="site_libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
  <script src="site_libs/jszip-1.10.20/jszip.min.js"></script>
  <link href="site_libs/dt-ext-buttons-1.10.20/css/buttons.dataTables.min.css" rel="stylesheet" />
  <script src="site_libs/dt-ext-buttons-1.10.20/js/dataTables.buttons.min.js"></script>
  <script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.flash.min.js"></script>
  <script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.html5.min.js"></script>
  <script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.colVis.min.js"></script>
  <script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.print.min.js"></script>
  <link href="site_libs/nouislider-7.0.10/jquery.nouislider.min.css" rel="stylesheet" />
  <script src="site_libs/nouislider-7.0.10/jquery.nouislider.min.js"></script>
  <link href="site_libs/selectize-0.12.0/selectize.bootstrap3.css" rel="stylesheet" />
  <script src="site_libs/selectize-0.12.0/selectize.min.js"></script>
  <link href="site_libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
  <script src="site_libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>
  <script src="site_libs/bowser-1.9.3/bowser.min.js"></script>
  <script src="site_libs/webcomponents-2.0.0/webcomponents.js"></script>
  <script src="site_libs/distill-2.2.21/template.v2.js"></script>
  <!--radix_placeholder_site_in_header-->
  <style type="text/css">
  /* Whole document: */
  @import url('https://fonts.googleapis.com/css?family=PT+Sans|PT+Serif|Lato:300|Roboto:300|Nunito+Sans:300|Inconsolata|Source+Sans+Pro|Work+Sans:200,300,400,500');
  /* controls general properties of site*/
  body {
    color: #414141;
  /*  line-height: 1.7;
    font-size: 17px; */
    background-color: #FEFEFE;
  }

  d-title h1{
  font-size: 1.8em;
  font-family: 'Work Sans';
  font-weight: 300;
  }

  d-article h1 {
    font-size: 1.8em;
    font-family: 'Work Sans', 'Source Sans Pro', 'PT Sans';
    font-weight: 300;
  }

  d-article h2 {
    font-size: 1.65em;
    font-family: 'Work Sans', 'Source Sans Pro', 'PT Sans';
    font-weight: 500;
  }

  d-article h3 {
    margin-top: 1em;
  }

  d-article h4 {
    font-size: 1em;
  }

  d-byline .authors-affiliations {
    grid-column-end: span 3;
    grid-column-gap: 8px;
  }

  a.name:-webkit-any-link {
    text-decoration: underline;
    text-decoration-style: dotted;
  }

  a.affiliation:-webkit-any-link {
    text-decoration: underline;
    text-decoration-style: dotted;
  }

  blockquote p {
  	font-size: 1.3em;
  	font-family: 'Lato';
    font-style: normal;
    margin: 0.5em 0 0.5em;
  }

  d-code {
    background-color: #E8E8E8;
    border-radius: 3px;
    font-size: 0.8em;
    font-family: 'Monaco', 'PT Mono', monospace;
    overflow-x: scroll !important;
    margin-bottom: 20px;
    border-left: 5px solid #323232;
    border-top: 1px solid #B7B7B7;
    border-right: 1px solid #B7B7B7;
    border-bottom: 1px solid #B7B7B7;
  }

  code {
    background-color: #F5F5F5;
    border-radius: 3px;
    font-size: 0.8em;
    font-family: 'Monaco', 'PT Mono', monospace;
    overflow-x: scroll !important;
    margin-bottom: 20px;
    color: #BF2727;
  }

  pre.text-output {
    margin-top: 0px;
     color: #333333;
     background-color: #FAFAFA;
     font-size: 0.8em
   }

  pre {
    border: 1px solid #ccc;
    padding: 0px 10px 10px 18px;
    margin: 1em 0px;
    border-radius: 5px;
  }

  .nav-dropdown-content .nav-dropdown-header {
    text-transform: none;
  }

  d-article li {
    margin: 0px 0px 4px;
  }

  /*************************************************
  *  code for side by side krona plot
  **************************************************/

  .column {
    float: left;
    width: 48%;
    padding: 5px;
  }

  /* Clear floats after image containers */
  .row::after {
    content: "";
    clear: both;
    display: table;
  }

  div.krona {
    width: 100%;
  	height: 90%;
    background-color: white;
    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
    margin-bottom: 25px;
  	margin-top: 5px;
  }

  div.container-krona {
    text-align: center;
    padding: 1px 10px;
  	color: #414141;
  	font-size: 1.48em;
  	line-height: 0.8em;
  	font-family: 'Inconsolata';
  }

  div.container-krona p {
    margin: 20px;
  }

  div.krona img {
    border:1px solid white;
    margin: 25px;
  }

  img {
    margin: 25px;
  }

  /** LIGHTBOX MARKUP **/
  .thumbnail {
    max-width: 100%;
    position: relative;
  }

  .lightbox {
  	/** Default lightbox to hidden */
  	display: none;

  	/** Position and style */
  	position: fixed;
  	z-index: 999;
  	width: 100%;
  	height: 100%;
  	text-align: center;
  	top: 0;
  	left: 0;
  	background: rgba(0,0,0,0.8);
  }

  .lightbox img {
  	/** Pad the lightbox image */
  	position: fixed;
  	top: 0;
  	left: 0;
  	right: 0;
  	bottom: 0;
  	max-width: 0%;
  	max-height: 0%;
  	margin: auto;
  	box-sizing: border-box;
  }

  .lightbox:target {
  	/** Remove default browser outline */
  	outline: none;

  	/** Unhide lightbox **/
  	display: block;
  }

  .lightbox:target img {
  	max-height: 100%;
  	max-width: 100%;
  }

  .distill-site-header .logo img {
    margin-top: 15px;
    max-height: 30px;
    margin-bottom: 5px;
  }

  .distill-site-nav {
      color: #F5F5F5;
      background-color: #323232;
      font-size: 16px;
      font-weight: 300;
  }

  .article-banner {
    width: auto;
    height: auto;
    border: 0;
  }

  img[src="assets/banner.png"] {
    border: 0;
    padding: 0;
    width: 100%;
    margin: 100px 50px 0px 25px;
  }

  img.map, map area {
    outline: none;
  }

  img[src="assets/wf_map.png"] {
    border: 0;
    padding: 0;
    margin: 0px 0px 0px 0px;
  }

  </style>
  <!--/radix_placeholder_site_in_header-->

  <link rel="stylesheet" href="assets/styles.css" type="text/css"/>

</head>

<body>

<!--radix_placeholder_front_matter-->

<script id="distill-front-matter" type="text/json">
{"title":"No 1. Indicator Analysis","description":"Script to identify the core fish gut microbiome using Indicator Analysis.","authors":[]}
</script>

<!--/radix_placeholder_front_matter-->
<!--radix_placeholder_navigation_before_body-->
<header class="header header--fixed" role="banner">
<nav class="distill-site-nav distill-site-header">
<div class="nav-left">
<a class="logo" href="index.html">
<img src="assets/icon.png" alt="Logo"/>
</a>
<a href="index.html" class="title"></a>
</div>
<div class="nav-right">
<a href="index.html">
<i class="fa fa-home" aria-hidden="true"></i>
</a>
<a href="wf1.html">1. Indicator Analysis</a>
<a href="wf2.html">2. Taxa by Sample Fraction</a>
<a href="wf3.html">3. Subset Fish</a>
<a href="wf4.html">4. Alpha Diversity</a>
<a href="wf5.html">5. Beta Diversity</a>
<a href="wf6.html">6. Beta Dispersion</a>
<a href="wf7.html">7. PIME Analysis</a>
<a href="https://github.com/bocasbiome/web/">
<i class="fa fa-github fa-lg" aria-hidden="true"></i>
</a>
<a href="javascript:void(0);" class="nav-toggle">&#9776;</a>
</div>
</nav>
</header>
<!--/radix_placeholder_navigation_before_body-->
<!--radix_placeholder_site_before_body-->
<!--/radix_placeholder_site_before_body-->

<div class="d-title">
<h1>No 1. Indicator Analysis</h1>
<p><p>Script to identify the core fish gut microbiome using Indicator Analysis.</p></p>
</div>


<div class="d-article">
<h3 class="d-toc-header">Table of Contents</h3>
<nav class="d-toc" id="TOC">
<ul>
<li><a href="#data-preparation">Data Preparation</a></li>
<li><a href="#run-indicator-analysis-to-identify-core-taxa">Run Indicator Analysis to Identify Core Taxa</a></li>
<li><a href="#fish-core-summary">Fish Core Summary</a></li>
<li><a href="#fish-core-plots">Fish Core Plots</a></li>
<li><a href="#source-code">Source Code</a></li>
</ul>
</nav>
<hr class="d-toc-separator"/>
<h2 id="data-preparation">Data Preparation</h2>
<p>Here, we use unrarefied data to identify the core. First, we get <code>ps</code> object containing all unrarefied sample fractions (i.e. fish, water, &amp; potential prey items).</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
ps.slv.tree.mbio.bocas &lt;- readRDS(&quot;rdata/p1/ps_16S_capis_bocas_all_unrar.rds&quot;)
ps.slv.tree.mbio.bocas</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 10711 taxa and 127 samples ]
sample_data() Sample Data:       [ 127 samples by 11 sample variables ]
tax_table()   Taxonomy Table:    [ 10711 taxa by 11 taxonomic ranks ]
phy_tree()    Phylogenetic Tree: [ 10711 tips and 10710 internal nodes ]</code></pre>
</div>
<p>Here is a summary table of all samples.</p>
<p><br/></p>
<div class="layout-chunk" data-layout="l-body">

</div>
<p><br/></p>
<div class="layout-chunk" data-layout="l-page">
<div id="nsuxann9l3fmx3yh74pu" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="nsuxann9l3fmx3yh74pu">{"x":{"filter":"top","filterHTML":"<tr>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"disabled\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"disabled\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;inside bay&quot;,&quot;outside bay&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;disturbed&quot;,&quot;healthy&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;Inner bay&quot;,&quot;Inner bay disturbed&quot;,&quot;Outer bay&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;ALR&quot;,&quot;CCR&quot;,&quot;Dock station&quot;,&quot;PBL&quot;,&quot;PPR&quot;,&quot;PST&quot;,&quot;RNW&quot;,&quot;ROL&quot;,&quot;SCR&quot;,&quot;SIS&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;Environment&quot;,&quot;Fish&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;Algae&quot;,&quot;Coral&quot;,&quot;Fish&quot;,&quot;Softcoral&quot;,&quot;Sponge&quot;,&quot;Water&quot;,&quot;Zoanthus&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;Agaricia tenuifolia&quot;,&quot;Amphimedon compressa&quot;,&quot;Amphiroa&quot;,&quot;Antillogorgia bipinnata&quot;,&quot;blue rope&quot;,&quot;brown&quot;,&quot;Chaetodon capistratus&quot;,&quot;Chondrilla nucula&quot;,&quot;infauna Amphimedon compressa&quot;,&quot;infauna light blue encrusting&quot;,&quot;Knobby_searod_Eunicea?&quot;,&quot;light blue encrusting&quot;,&quot;orange&quot;,&quot;Palythoa caribaeorum&quot;,&quot;Poritis furcata&quot;,&quot;Siderastrea sidera&quot;,&quot;Turf&quot;,&quot;Water&quot;,&quot;Zoanthid&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"integer\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none; position: absolute; width: 200px;\">\n      <div data-min=\"10369\" data-max=\"79466\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n<\/tr>","extensions":["Buttons"],"caption":"<caption style=\"caption-side: bottom; text-align: left;\">\n  Table: \n  <em>Sample  summary table. Use the buttons to\n            navigate through the table or download a copy.<\/em>\n<\/caption>","data":[["ML1855","ML1856","ML1857","ML1861","ML1862","ML1866","ML1871","ML1875","ML1876","ML2360","ML2363","ML2364","ML2365","ML2366","ML2367","ML2368","ML2369","ML2370","ML2371","ML2372","ML2373","ML2374","ML2375","ML2376","ML2377","ML2379","ML2380","ML2382","ML2383","ML2384","ML2385","ML2388","ML2390","ML2392","ML2393","ML2394","ML2396","ML2397","ML2404","ML2405","ML2406","ML2407","ML2409","ML2410","ML2411","ML2412","ML2413","ML2414","ML2415","ML2416","ML2417","ML2418","ML2419","ML2421","ML2429","ML2432","ML2433","ML2438","ML2440","ML2441","ML2442","ML2445","ML2446","ML2447","ML2448","ML2449","ML2451","ML2452","ML2454","ML2455","ML2456","ML2459","ML2460","ML2463","ML2466","ML2468","ML2469","ML2471","ML2472","ML2473","ML2475","ML2478","ML2481","ML2486","ML2487","ML2488","ML2489","ML2490","ML2491","ML2492","ML2493","ML2494","ML2495","ML2496","ML2501","ML2502","ML2504","ML2505","ML2507","ML2509","ML2510","ML2511","ML2513","ML2514","ML2515","ML2521","ML2522","ML2526","ML2527","ML2529","ML2531","ML2532","ML2533","ML2534","ML2538","ML2541","ML2542","ML2543","ML2670","ML2671","ML2673","ML2674","ML2675","ML2677","ML2678","ML2680","ML2681"],["Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean","Panama caribbean"],["Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro","Bocas del Toro"],["inside bay","outside bay","inside bay","inside bay","inside bay","inside bay","outside bay","outside bay","inside bay","outside bay","inside bay","outside bay","inside bay","outside bay","inside bay","inside bay",null,null,"inside bay","inside bay","inside bay","outside bay","inside bay","inside bay","outside bay","outside bay","outside bay",null,"outside bay","outside bay","inside bay","inside bay","outside bay","inside bay","outside bay","outside bay","inside bay","outside bay","inside bay","outside bay","inside bay","outside bay","inside bay","inside bay","inside bay","inside bay","outside bay","inside bay","outside bay","inside bay","outside bay","outside bay","inside bay","outside bay","outside bay","inside bay","outside bay","inside bay","inside bay","inside bay","inside bay","inside bay","outside bay","inside bay","inside bay","outside bay","inside bay","inside bay","inside bay","inside bay","inside bay","outside bay","inside bay","inside bay","inside bay","inside bay","inside bay","outside bay","outside bay","inside bay","inside bay","outside bay","inside bay","inside bay","inside bay","outside bay","inside bay","inside bay","inside bay","outside bay","inside bay","inside bay","outside bay","inside bay","outside bay","inside bay","outside bay","inside bay","inside bay","inside bay","outside bay","outside bay","outside bay","outside bay","inside bay","inside bay","inside bay","outside bay","outside bay","outside bay","outside bay","inside bay","inside bay","inside bay","inside bay","outside bay","outside bay","inside bay","inside bay","inside bay","outside bay","outside bay","outside bay","inside bay","inside bay","inside bay","inside bay"],["disturbed","healthy","healthy","healthy","healthy","disturbed","healthy","healthy","disturbed","healthy","healthy","healthy","healthy","healthy","disturbed","healthy",null,null,"healthy","disturbed","healthy","healthy","disturbed","healthy","healthy","healthy","healthy",null,"healthy","healthy","healthy","disturbed","healthy","healthy","healthy","healthy","healthy","healthy","healthy","healthy","disturbed","healthy","healthy","disturbed","healthy","healthy","healthy","healthy","healthy","healthy","healthy","healthy","healthy","healthy","healthy","healthy","healthy","disturbed","healthy","disturbed","healthy","healthy","healthy","healthy","healthy","healthy","healthy","healthy","healthy","healthy","disturbed","healthy","disturbed","healthy","disturbed","disturbed","healthy","healthy","healthy","healthy","disturbed","healthy","disturbed","healthy","healthy","healthy","healthy","healthy","healthy","healthy","disturbed","healthy","healthy","disturbed","healthy","disturbed","healthy","healthy","healthy","healthy","healthy","healthy","healthy","healthy","disturbed","healthy","disturbed","healthy","healthy","healthy","healthy","healthy","disturbed","healthy","disturbed","healthy","healthy","disturbed","disturbed","disturbed","healthy","healthy","healthy","healthy","healthy","disturbed","disturbed"],["Inner bay disturbed","Outer bay","Inner bay","Inner bay","Inner bay","Inner bay disturbed","Outer bay","Outer bay","Inner bay disturbed","Outer bay","Inner bay","Outer bay","Inner bay","Outer bay","Inner bay disturbed","Inner bay",null,null,"Inner bay","Inner bay disturbed","Inner bay","Outer bay","Inner bay disturbed","Inner bay","Outer bay","Outer bay","Outer bay",null,"Outer bay","Outer bay","Inner bay","Inner bay disturbed","Outer bay","Inner bay","Outer bay","Outer bay","Inner bay","Outer bay","Inner bay","Outer bay","Inner bay disturbed","Outer bay","Inner bay","Inner bay disturbed","Inner bay","Inner bay","Outer bay","Inner bay","Outer bay","Inner bay","Outer bay","Outer bay","Inner bay","Outer bay","Outer bay","Inner bay","Outer bay","Inner bay disturbed","Inner bay","Inner bay disturbed","Inner bay","Inner bay","Outer bay","Inner bay","Inner bay","Outer bay","Inner bay","Inner bay","Inner bay","Inner bay","Inner bay disturbed","Outer bay","Inner bay disturbed","Inner bay","Inner bay disturbed","Inner bay disturbed","Inner bay","Outer bay","Outer bay","Inner bay","Inner bay disturbed","Outer bay","Inner bay disturbed","Inner bay","Inner bay","Outer bay","Inner bay","Inner bay","Inner bay","Outer bay","Inner bay disturbed","Inner bay","Outer bay","Inner bay disturbed","Outer bay","Inner bay disturbed","Outer bay","Inner bay","Inner bay","Inner bay","Outer bay","Outer bay","Outer bay","Outer bay","Inner bay disturbed","Inner bay","Inner bay disturbed","Outer bay","Outer bay","Outer bay","Outer bay","Inner bay","Inner bay disturbed","Inner bay","Inner bay disturbed","Outer bay","Outer bay","Inner bay disturbed","Inner bay disturbed","Inner bay disturbed","Outer bay","Outer bay","Outer bay","Inner bay","Inner bay","Inner bay disturbed","Inner bay disturbed"],["PST","SCR","SIS","ROL","ALR","PBL","CCR","PPR","RNW","SCR","ALR","PPR","ALR","PPR","PBL","ALR",null,null,"ALR","Dock station","ALR","PPR","Dock station","ALR","PPR","SCR","SCR",null,"SCR","CCR","ROL","RNW","CCR","ALR","SCR","CCR","ALR","SCR","SIS","CCR","PBL","PPR","ROL","RNW","SIS","ROL","SCR","ROL","SCR","SIS","SCR","CCR","ROL","PPR","CCR","ROL","SCR","RNW","ALR","PST","ALR","ALR","SCR","ALR","ROL","CCR","SIS","SIS","ROL","ALR","RNW","CCR","PBL","ROL","PST","PBL","SIS","CCR","SCR","ROL","PBL","CCR","PST","ALR","ROL","CCR","ALR","ALR","ROL","PPR","PBL","ALR","SCR","PST","CCR","RNW","CCR","ROL","ROL","ALR","CCR","PPR","SCR","CCR","RNW","SIS","RNW","SCR","PPR","CCR","SCR","SIS","PBL","ROL","RNW","PPR","PPR","PST","PST","PBL","SCR","PPR","CCR","ROL","ALR","PST","PBL"],["Environment","Environment","Environment","Environment","Environment","Environment","Environment","Environment","Environment","Environment","Environment","Environment","Environment","Environment","Environment","Environment","Environment","Environment","Environment","Environment","Environment","Environment","Environment","Environment","Environment","Environment","Environment","Environment","Environment","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Environment","Environment","Environment","Environment","Environment","Environment","Environment","Environment","Environment"],["Water","Water","Water","Water","Water","Water","Water","Water","Water","Algae","Algae","Sponge","Sponge","Sponge","Sponge","Sponge","Sponge","Sponge","Sponge","Coral","Coral","Coral","Coral","Coral","Coral","Softcoral","Softcoral","Zoanthus","Zoanthus","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Fish","Water","Water","Water","Water","Water","Water","Water","Water","Water"],["Water","Water","Water","Water","Water","Water","Water","Water","Water","Amphiroa","Turf","blue rope","orange","brown","Chondrilla nucula","light blue encrusting","Amphimedon compressa","infauna Amphimedon compressa","infauna light blue encrusting","Siderastrea sidera","Poritis furcata","Agaricia tenuifolia","Siderastrea sidera","Poritis furcata","Agaricia tenuifolia","Antillogorgia bipinnata","Knobby_searod_Eunicea?","Zoanthid","Palythoa caribaeorum","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Chaetodon capistratus","Water","Water","Water","Water","Water","Water","Water","Water","Water"],[46686,61553,38398,43066,45139,58720,52288,57095,53664,26886,45359,37808,35929,40828,38573,58711,35405,33825,40614,46708,14432,45017,37442,10369,36006,11555,54633,29975,46056,43119,35389,25485,75913,46315,32132,44017,38071,42558,24500,34459,35613,32445,53475,40765,42158,36603,34871,26583,60608,44778,45546,39384,51943,70447,47790,31161,49370,35881,48357,34677,30401,38418,30896,33080,35176,41694,40092,53629,39658,33041,35737,43411,43193,38949,47336,15754,49856,47325,31635,47136,40286,41437,41577,51927,58098,40508,39139,41801,45290,31830,25329,37731,41570,43056,44984,34308,42900,41441,34865,37957,28842,45242,18533,37027,53875,79466,38861,35623,43540,34657,25673,41487,42202,35662,36708,34824,34825,53514,56971,46559,48512,52118,51878,44815,46067,49624,49278]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Sample_ID<\/th>\n      <th>Region<\/th>\n      <th>Locality<\/th>\n      <th>Position<\/th>\n      <th>Reef_type<\/th>\n      <th>Zone<\/th>\n      <th>Reef<\/th>\n      <th>Fraction<\/th>\n      <th>Fraction_delailed<\/th>\n      <th>Species<\/th>\n      <th>SeqDepth<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"dom":"Blfrtip","buttons":["copy","csv","excel"],"pageLength":5,"lengthMenu":[[5,10,50,-1],["5","10","50","All"]],"columnDefs":[{"className":"dt-right","targets":10}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true,"rowCallback":"function(row, data) {\nvar value=data[0]; $(this.api().cell(row, 0).node()).css({'font-size':'80%'});\nvar value=data[1]; $(this.api().cell(row, 1).node()).css({'font-size':'80%'});\nvar value=data[2]; $(this.api().cell(row, 2).node()).css({'font-size':'80%'});\nvar value=data[3]; $(this.api().cell(row, 3).node()).css({'font-size':'80%'});\nvar value=data[4]; $(this.api().cell(row, 4).node()).css({'font-size':'80%'});\nvar value=data[5]; $(this.api().cell(row, 5).node()).css({'font-size':'80%'});\nvar value=data[6]; $(this.api().cell(row, 6).node()).css({'font-size':'80%'});\nvar value=data[7]; $(this.api().cell(row, 7).node()).css({'font-size':'80%'});\nvar value=data[8]; $(this.api().cell(row, 8).node()).css({'font-size':'80%'});\nvar value=data[9]; $(this.api().cell(row, 9).node()).css({'font-size':'80%'});\nvar value=data[10]; $(this.api().cell(row, 10).node()).css({'font-size':'80%'});\n}"}},"evals":["options.rowCallback"],"jsHooks":[]}</script>
</div>
<p><br/></p>
<p>Next, we make a column in taxonomy table for ASV IDs by number. <em>Caution</em>, ASV ID numbers will differ between the core (unrarefied) and the rarefied data set.</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
tax_table(ps.slv.tree.mbio.bocas) &lt;- cbind(tax_table(ps.slv.tree.mbio.bocas),
                                           rownames(tax_table(ps.slv.tree.mbio.bocas)))

colnames(tax_table(ps.slv.tree.mbio.bocas)) &lt;-
        c(&quot;Kingdom&quot;, &quot;Phylum&quot;, &quot;Class&quot;,
          &quot;Order&quot;, &quot;Family&quot;, &quot;Genus&quot;, &quot;ASV&quot;)</code></pre>
</div>
<p>Next, we change row names in taxonomy table to AVS IDs for plotting purposes.</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
taxa_names(ps.slv.tree.mbio.bocas) &lt;- 1:ntaxa(ps.slv.tree.mbio.bocas)
taxa_names(ps.slv.tree.mbio.bocas) &lt;- paste(&quot;ASV_&quot;,
                                            taxa_names(ps.slv.tree.mbio.bocas),
                                            sep = &quot;&quot;) #JJS# Changed</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
tax.unrar.bocas.all &lt;- data.frame(tax_table(ps.slv.tree.mbio.bocas))
tax.unrar.new &lt;- data.frame(ASV_ID = row.names(tax.unrar.bocas.all),
                            tax.unrar.bocas.all)
tax.unrar.new &lt;- tax.unrar.new[, c(2,3,4,5,6,7,1,8)]
tax.unrar.new2 &lt;- as.matrix(tax.unrar.new)
ps.slv.tree.mbio.bocas &lt;- merge_phyloseq(otu_table(ps.slv.tree.mbio.bocas),
                                         tax_table(tax.unrar.new2),
                                         sample_data(ps.slv.tree.mbio.bocas),
                                         phy_tree(ps.slv.tree.mbio.bocas))
saveRDS(ps.slv.tree.mbio.bocas, &quot;rdata/p1/ps_16S_bocas_unrar_all_ASVID.rds&quot;)</code></pre>
</div>
<p>We will also rename all <code>NA</code> taxa by the next highest named rank. To do this we first, create a copy of the original <code>ps</code> object <em>before</em> renaming taxa. That way we have a copy for other analyses.</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
ps.slv.tree.mbio.bocas_o &lt;- ps.slv.tree.mbio.bocas</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
tax.clean &lt;- data.frame(tax_table(ps.slv.tree.mbio.bocas))
for (i in 1:6){ tax.clean[,i] &lt;- as.character(tax.clean[,i])}
tax.clean[is.na(tax.clean)] &lt;- &quot;&quot;

for (i in 1:nrow(tax.clean)){
    if (tax.clean[i,2] == &quot;&quot;){
        kingdom &lt;- base::paste(&quot;k&quot;, tax.clean[i,1], sep = &quot;_&quot;)
        tax.clean[i, 2:6] &lt;- kingdom
    } else if (tax.clean[i,3] == &quot;&quot;){
        phylum &lt;- base::paste(&quot;p&quot;, tax.clean[i,2], sep = &quot;_&quot;)
        tax.clean[i, 3:6] &lt;- phylum
    } else if (tax.clean[i,4] == &quot;&quot;){
        class &lt;- base::paste(&quot;c&quot;, tax.clean[i,3], sep = &quot;_&quot;)
        tax.clean[i, 4:6] &lt;- class
    } else if (tax.clean[i,5] == &quot;&quot;){
        order &lt;- base::paste(&quot;o&quot;, tax.clean[i,4], sep = &quot;_&quot;)
        tax.clean[i, 5:6] &lt;- order
    } else if (tax.clean[i,6] == &quot;&quot;){
        tax.clean$Genus[i] &lt;- base::paste(&quot;f&quot;,tax.clean$Family[i], sep = &quot;_&quot;)
        }
}
rm(class, order, phylum, kingdom)</code></pre>
</div>
<p>Then, we create new ASV names that have lowest rank name attached.</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
tax.clean &lt;- tax.clean %&gt;% unite(&quot;ASV_IDa&quot;, Genus:ASV_ID,
                                 remove = FALSE, sep = &quot;_&quot;)
tax.clean &lt;- tax.clean %&gt;% unite(&quot;ASV_IDb&quot;, ASV_ID:Genus,
                                 remove = FALSE, sep = &quot;_&quot;)
tax.clean &lt;- tax.clean[, c(1,2,3,4,5,8,9,6,7,10)]
tax.clean$ASV_IDa &lt;-
  str_replace_all(tax.clean$ASV_IDa,
                  &#39;Clostridium_sensu_stricto_[0-9]&#39;,
                  &#39;Clostridium&#39;)
tax.clean$ASV_IDb &lt;-
  str_replace_all(tax.clean$ASV_IDb,
                  &#39;Clostridium_sensu_stricto_[0-9]&#39;,
                  &#39;Clostridium&#39;)
tax.clean$ASV_IDc &lt;- tax.clean$ASV_IDa
tax.clean$ASV_IDc &lt;-
  str_replace_all(tax.clean$ASV_IDc,
                  &#39;_ASV&#39;, &#39;&#39;)
tax.clean &lt;- tax.clean[, c(1,2,3,4,5,6,7,8,9,11,10)]
write.csv(tax.clean, &quot;tables/p1/tax_all_new_no_na.csv&quot;)</code></pre>
</div>
<p>And finally, add the taxonomy table back to the phyloseq object.</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
tax_table(ps.slv.tree.mbio.bocas) &lt;- as.matrix(tax.clean)
rank_names(ps.slv.tree.mbio.bocas)
saveRDS(ps.slv.tree.mbio.bocas, &quot;rdata/p1/ps_16S_bocas_unrar_all_ASVID_no_NA.rds&quot;)</code></pre>
</div>
<h2 id="run-indicator-analysis-to-identify-core-taxa">Run Indicator Analysis to Identify Core Taxa</h2>
<p>Now we can run the Indicator Analysis.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Step one is to generate an ASV table without row names.</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
data.IndVal.env.fish &lt;- data.frame(otu_table(ps.slv.tree.mbio.bocas))
data.IndVal.env.fish
data.IndVal.ASV &lt;- tibble::remove_rownames(data.IndVal.env.fish)</code></pre>
</div>
<p>And then generate an Indicator value group file.</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
data.IndVal.group &lt;- data.frame(sample_data(ps.slv.tree.mbio.bocas)) %&gt;%
  select(Fraction)
data.IndVal.group$Status &lt;- data.IndVal.group$Fraction
data.IndVal.group &lt;- data.IndVal.group[, c(2,1)]
data.IndVal.group$Status &lt;- str_replace(data.IndVal.group$Status, &quot;Environment&quot;, &quot;1&quot;)
data.IndVal.group$Status &lt;- str_replace(data.IndVal.group$Status, &quot;Fish&quot;, &quot;2&quot;)
data.IndVal.group &lt;- tibble::rownames_to_column(data.IndVal.group, &quot;Label&quot;)
data.IndVal.group$Status &lt;-  as.integer(data.IndVal.group$Status)
data.IndVal.group$Fraction &lt;-  as.character(data.IndVal.group$Fraction)</code></pre>
</div>
<p>Next, calculate the indicator values. We set a seed for reproducibility and saved a table of results.</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
set.seed(1280)
iva &lt;- indval(data.IndVal.ASV, data.IndVal.group$Status)
#Table of the significant indicator species at p= 0.01
gr &lt;- iva$maxcls[iva$pval &lt;= 0.01]
iv &lt;- iva$indcls[iva$pval &lt;= 0.01]
pv &lt;- iva$pval[iva$pval &lt;= 0.01]
fr &lt;- apply(data.IndVal.ASV &gt; 0, 2, sum)[iva$pval &lt;= 0.01]
indval.out &lt;- data.frame(group = gr, indval = iv, pval = pv, freq = fr)
indval.out &lt;- indval.out[order(indval.out$group, -indval.out$indval),]
indval.out
write.csv(indval.out,
          file = &quot;tables/p1/IndVal_microbiome_env_fish_output_p01_new0.csv&quot;)</code></pre>
</div>
<p>Finally, we correct p-values for repeated testing and save a new table of corrected results.</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
indval.out.prob.corrected = p.adjust(indval.out$pval, &quot;bonferroni&quot;)
write.csv(indval.out.prob.corrected,
          file = &quot;tables/p1/IndVal_microbiome_env_fish_bonf_new0.csv&quot;)</code></pre>
</div>
<h2 id="fish-core-summary">Fish Core Summary</h2>
<p>Now we plot the results of the <strong>fish core</strong> by subsetting only the ASVs that were significant for fish. First, we grab the ASVs from the phyloseq object.</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
fish_ind_asvs &lt;- indval.out[indval.out$group == 2, ]
fish_ind_asvs_list &lt;- row.names(fish_ind_asvs)
ps.indv.core.01 &lt;- subset_taxa(ps.slv.tree.mbio.bocas,
                               rownames(tax_table(ps.slv.tree.mbio.bocas)) %in%
                                       fish_ind_asvs_list)
ps.indv.core.01</code></pre>
</div>
<p>Then pull out the fish-only samples.</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
ps.indv.core.fish &lt;- subset_samples(ps.indv.core.01, Fraction == &quot;Fish&quot;)
sample_data(ps.indv.core.fish)
tax.core &lt;- tax_table(ps.indv.core.fish)

write.csv(tax.core, file = &quot;tables/p1/tax_indVal_core.csv&quot;)
tax_core_mod &lt;- read.csv(&quot;tables/p1/tax_indVal_core.csv&quot;, header = TRUE, row.names = 1)
tax_core_mod2 &lt;- as.matrix(tax_core_mod)</code></pre>
</div>
<p>And add the modified tax table to ps object and save ps object. This contains all ASVs identified with Indicator analysis at 0.01 for fish gut samples.</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
ps.indv.core.fish &lt;- merge_phyloseq(otu_table(ps.indv.core.fish),
                                    tax_table(tax_core_mod2),
                                    sample_data(ps.indv.core.fish),
                                    phy_tree(ps.indv.core.fish))
saveRDS(ps.indv.core.fish, &quot;rdata/p1/ps_indv01_core_fish.rds&quot;)
ps.indv.core.fish &lt;- readRDS(&quot;rdata/p1/ps_indv01_core_fish.rds&quot;)</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 27 taxa and 89 samples ]
sample_data() Sample Data:       [ 89 samples by 11 sample variables ]
tax_table()   Taxonomy Table:    [ 27 taxa by 11 taxonomic ranks ]
phy_tree()    Phylogenetic Tree: [ 27 tips and 26 internal nodes ]</code></pre>
</div>
<p>Here is a summary of the data set.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<table>
<thead>
<tr class="header">
<th>Metric</th>
<th>Results</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Min. number of reads</td>
<td>326</td>
</tr>
<tr class="even">
<td>Max. number of reads</td>
<td>63937</td>
</tr>
<tr class="odd">
<td>Total number of reads</td>
<td>2775814</td>
</tr>
<tr class="even">
<td>Average number of reads</td>
<td>31189</td>
</tr>
<tr class="odd">
<td>Median number of reads</td>
<td>33658</td>
</tr>
<tr class="even">
<td>Sparsity</td>
<td>0.513</td>
</tr>
<tr class="odd">
<td>Any ASVs sum to 1 or less?</td>
<td>FALSE</td>
</tr>
<tr class="even">
<td>Number of singleton ASVs</td>
<td>NA</td>
</tr>
<tr class="odd">
<td>Percent of ASVs that are singletons</td>
<td>NA</td>
</tr>
</tbody>
</table>
<p>And a look at the individual ASVs.</p>
<p><br/></p>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-page">
<div id="nsuxann9l3fmx3yh74pu" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="nsuxann9l3fmx3yh74pu">{"x":{"filter":"top","filterHTML":"<tr>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"disabled\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none; position: absolute; width: 200px;\">\n      <div data-min=\"0.213483146067415\" data-max=\"0.996349277001286\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none; position: absolute; width: 200px;\">\n      <div data-min=\"0.001\" data-max=\"0.009\" data-scale=\"3\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"integer\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none; position: absolute; width: 200px;\">\n      <div data-min=\"19\" data-max=\"111\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n<\/tr>","extensions":["Buttons"],"caption":"<caption style=\"caption-side: bottom; text-align: left;\">\n  Table: \n  <em>InVal  summary table. Use the buttons to\n            navigate through the table or download a copy.<\/em>\n<\/caption>","data":[["ASV_1","ASV_2","ASV_3","ASV_5","ASV_6","ASV_7","ASV_9","ASV_10","ASV_11","ASV_14","ASV_15","ASV_17","ASV_18","ASV_19","ASV_24","ASV_25","ASV_27","ASV_30","ASV_39","ASV_41","ASV_59","ASV_68","ASV_74","ASV_94","ASV_95","ASV_163","ASV_589"],["Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria"],["Proteobacteria","Spirochaetes","Proteobacteria","Proteobacteria","Proteobacteria","Proteobacteria","Firmicutes","Firmicutes","Proteobacteria","Firmicutes","Firmicutes","Proteobacteria","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Proteobacteria","Proteobacteria","Firmicutes","Firmicutes","Proteobacteria","Proteobacteria","Deinococcus-Thermus"],["Gammaproteobacteria","Spirochaetia","Gammaproteobacteria","Gammaproteobacteria","Gammaproteobacteria","Gammaproteobacteria","Clostridia","Clostridia","Gammaproteobacteria","Clostridia","Clostridia","Gammaproteobacteria","Clostridia","Clostridia","Clostridia","Clostridia","Clostridia","Clostridia","Clostridia","Clostridia","Gammaproteobacteria","Gammaproteobacteria","Clostridia","Clostridia","Gammaproteobacteria","Gammaproteobacteria","Deinococci"],["Oceanospirillales","Brevinematales","Oceanospirillales","Oceanospirillales","Oceanospirillales","Oceanospirillales","Clostridiales","Clostridiales","Oceanospirillales","Clostridiales","Clostridiales","Oceanospirillales","Clostridiales","Clostridiales","Clostridiales","Clostridiales","Clostridiales","Clostridiales","Clostridiales","Clostridiales","Oceanospirillales","Oceanospirillales","Clostridiales","Clostridiales","Vibrionales","Oceanospirillales","Thermales"],["Endozoicomonadaceae","Brevinemataceae","Endozoicomonadaceae","Endozoicomonadaceae","Endozoicomonadaceae","Endozoicomonadaceae","Ruminococcaceae","Lachnospiraceae","Endozoicomonadaceae","Ruminococcaceae","Ruminococcaceae","Endozoicomonadaceae","Ruminococcaceae","Clostridiaceae_1","Lachnospiraceae","Ruminococcaceae","Lachnospiraceae","Peptostreptococcaceae","Ruminococcaceae","Lachnospiraceae","Endozoicomonadaceae","Endozoicomonadaceae","Clostridiaceae_1","Peptostreptococcaceae","Vibrionaceae","Endozoicomonadaceae","Thermaceae"],["Endozoicomonas","Brevinema","Endozoicomonas","Endozoicomonas","Endozoicomonas","Endozoicomonas","Flavonifractor","f_Lachnospiraceae","Endozoicomonas","f_Ruminococcaceae","f_Ruminococcaceae","Endozoicomonas","f_Ruminococcaceae","Clostridium_sensu_stricto_1","Tyzzerella","f_Ruminococcaceae","Epulopiscium","Romboutsia","Anaerofilum","Epulopiscium","Endozoicomonas","Endozoicomonas","Clostridium_sensu_stricto_2","Romboutsia","Vibrio","Endozoicomonas","Thermus"],[0.996349277001286,0.659837501810194,0.538432561035095,0.741301700103198,0.730337078651685,0.696629213483146,0.707730186569061,0.550260717110305,0.674157303370786,0.696629213483146,0.426966292134831,0.539325842696629,0.572730392609601,0.314606741573034,0.314606741573034,0.325842696629214,0.50541953726853,0.370786516853933,0.213483146067416,0.269662921348315,0.235955056179775,0.447638937869512,0.258426966292135,0.345517530137765,0.348314606741573,0.251085003836027,0.235955056179775],[0.001,0.001,0.001,0.001,0.001,0.001,0.001,0.001,0.001,0.001,0.001,0.001,0.001,0.003,0.001,0.004,0.002,0.003,0.009,0.009,0.006,0.001,0.008,0.004,0.002,0.005,0.006],[111,71,68,67,65,62,65,53,60,62,38,48,52,28,28,29,46,33,19,24,21,41,23,32,31,24,21]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>ID<\/th>\n      <th>Kingdom<\/th>\n      <th>Phylum<\/th>\n      <th>Class<\/th>\n      <th>Order<\/th>\n      <th>Family<\/th>\n      <th>Genus<\/th>\n      <th>indval<\/th>\n      <th>pval<\/th>\n      <th>freq<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"dom":"Blfrtip","buttons":["copy","csv","excel"],"pageLength":5,"lengthMenu":[[5,10,-1],["5","10","All"]],"columnDefs":[{"targets":7,"render":"function(data, type, row, meta) { return DTWidget.formatRound(data, 3, 3, \",\", \".\"); }"},{"targets":8,"render":"function(data, type, row, meta) { return DTWidget.formatRound(data, 3, 3, \",\", \".\"); }"},{"className":"dt-right","targets":[7,8,9]}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true,"rowCallback":"function(row, data) {\nvar value=data[0]; $(this.api().cell(row, 0).node()).css({'font-size':'80%'});\nvar value=data[1]; $(this.api().cell(row, 1).node()).css({'font-size':'80%'});\nvar value=data[2]; $(this.api().cell(row, 2).node()).css({'font-size':'80%'});\nvar value=data[3]; $(this.api().cell(row, 3).node()).css({'font-size':'80%'});\nvar value=data[4]; $(this.api().cell(row, 4).node()).css({'font-size':'80%'});\nvar value=data[5]; $(this.api().cell(row, 5).node()).css({'font-size':'80%'});\nvar value=data[6]; $(this.api().cell(row, 6).node()).css({'font-size':'80%'});\nvar value=data[7]; $(this.api().cell(row, 7).node()).css({'font-size':'80%'});\nvar value=data[8]; $(this.api().cell(row, 8).node()).css({'font-size':'80%'});\nvar value=data[9]; $(this.api().cell(row, 9).node()).css({'font-size':'80%'});\n}"}},"evals":["options.columnDefs.0.render","options.columnDefs.1.render","options.rowCallback"],"jsHooks":[]}</script>
</div>
<p><br/></p>
<div class="layout-chunk" data-layout="l-body">

</div>
<h2 id="fish-core-plots">Fish Core Plots</h2>
<p>First, we set all of the plot parameters</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
ASV_colors &lt;- c(
        &quot;mediumaquamarine&quot;,&quot;darkcyan&quot;, &quot;darkseagreen2&quot;,
        &quot;springgreen4&quot;, &quot;lightseagreen&quot;,&quot;aquamarine2&quot;,
        &quot;turquoise&quot;, &quot;aquamarine&quot;,&quot;aquamarine4&quot;, &quot;darkseagreen4&quot;,
        &quot;dodgerblue1&quot;,&quot;dodgerblue3&quot;, &quot;lightsalmon3&quot;, &quot;coral&quot;,
        &quot;coral2&quot;,&quot;darksalmon&quot;,&quot;lightsalmon&quot;,&quot;coral3&quot;, &quot;salmon3&quot; ,
        &quot;indianred&quot;, &quot;indianred2&quot;,&quot;salmon2&quot;, &quot;salmon&quot;,&quot;lightsalmon4&quot;,
        &quot;lavender&quot;,&quot;darkslategrey&quot;,&quot;plum3&quot;)

level_order &lt;- c(&#39;Outer bay&#39;,
                 &#39;Inner bay&#39;,
                 &#39;Inner bay disturbed&#39;)
level_order2 &lt;- c(&#39;SCR&#39;, &#39;PPR&#39;, &#39;CCR&#39;, &#39;ALR&#39;,
                  &#39;SIS&#39;,&#39;ROL&#39;,&#39;RNW&#39;, &#39;PST&#39;, &#39;PBL&#39;)
asv_order &lt;- rev(c(&quot;ASV_95&quot;, &quot;ASV_589&quot;, &quot;ASV_2&quot;, &quot;ASV_25&quot;, &quot;ASV_18&quot;,
               &quot;ASV_15&quot;, &quot;ASV_14&quot;, &quot;ASV_94&quot;, &quot;ASV_30&quot;, &quot;ASV_10&quot;,
               &quot;ASV_9&quot;, &quot;ASV_24&quot;, &quot;ASV_74&quot;, &quot;ASV_19&quot;, &quot;ASV_39&quot;,
               &quot;ASV_41&quot;, &quot;ASV_27&quot;, &quot;ASV_7&quot;, &quot;ASV_68&quot;, &quot;ASV_6&quot;,
               &quot;ASV_59&quot;, &quot;ASV_5&quot;, &quot;ASV_3&quot;, &quot;ASV_17&quot;, &quot;ASV_163&quot;,
               &quot;ASV_11&quot;, &quot;ASV_1&quot;))</code></pre>
</div>
<p>Then we agglomerate at phylum level and transform read counts to relative abundance.</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
capis.core &lt;- ps.indv.core.fish %&gt;%
        tax_glom(taxrank = &quot;ASV_IDa&quot;) %&gt;%
        transform_sample_counts(function(x) {x/sum(x)} ) %&gt;%
        psmelt() %&gt;%
        filter(Abundance &gt; 0.00) %&gt;%
        arrange(ASV_IDa)
capis.core$ASV_ID &lt;- gdata::reorder.factor(capis.core$ASV_ID, new.order = asv_order)
capis.core &lt;- capis.core %&gt;% dplyr::arrange(ASV_ID)
capis.core$ASV_IDa &lt;- factor(capis.core$ASV_IDa, levels = unique(capis.core$ASV_IDa))
levels(capis.core$ASV_IDa)
attributes(capis.core$ASV_IDa)</code></pre>
</div>
<p>Plot first figure.</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
fig01 &lt;- ggplot(capis.core, aes(x = factor(Zone, level = level_order),
                                y = Abundance, fill = ASV_IDa)) +
        geom_bar(stat = &quot;identity&quot;, position = &quot;fill&quot;) +
        scale_fill_manual(values = ASV_colors) +
        scale_x_discrete(&quot;Zone&quot;, expand = waiver(),
                         position = &quot;bottom&quot;, drop = FALSE) +
        guides(fill = guide_legend(reverse = TRUE,
                                   keywidth = 1,
                                   keyheight = 1,
                                   title=&quot;ASV ID&quot;)) + #JJS# added new title
        ylab(&quot;&quot;) +
        ggtitle(&quot;&quot;, subtitle = &quot;Core across zones&quot;) +
        theme(plot.title = element_text(size = 18)) +
        theme(plot.subtitle = element_text(size = 16)) +
        theme_cowplot()</code></pre>
</div>
<p>Plot second figure.</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
fig02 &lt;-
        ggplot(capis.core, aes(x = Species, y = Abundance, fill = ASV_IDa)) +
        geom_bar(stat = &quot;identity&quot;, position = &quot;fill&quot;) +
        scale_fill_manual(values = ASV_colors) +
        scale_x_discrete(&quot;&quot;, expand = waiver(),
                         position = &quot;bottom&quot;, drop = FALSE) +
        guides(fill = guide_legend(reverse = TRUE,
                                   keywidth = 1,
                                   keyheight = 1,
                                   title = &quot;ASV ID&quot;)) + #JJS# added new title
        ylab(&quot;Relative Abundance (ASV)&quot;) +
        ggtitle(&quot;Fish gut core microbiome&quot;, subtitle = &quot;&quot;) +
        theme(plot.title = element_text(size = 18)) +
        theme_cowplot()</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-page">
<p><img src="wf1_files/figure-html5/unnamed-chunk-30-1.png" width="624" /></p>
</div>
<p>That’s the end of Script 1. In the next Script we compare taxonomic composition across different sample fractions.</p>
<p></br></p>
<div class="post-nav">
<div class="post-nav-item">
<div class="meta-nav">
Next
</div>
<p><a href="wf2.html" rel="next">N<sup><u>o</u></sup> 2 Comparing Sample Fractions</a></p>
</div>
</div>
<h2 id="source-code" class="appendix">Source Code</h2>
<p>The source code for this page can be accessed on GitHub by <a href="https://github.com/bocasbiome/web/blob/master/wf1.Rmd">clicking this link</a>.</p>
<!--radix_placeholder_article_footer-->
<!--/radix_placeholder_article_footer-->
</div>

<div class="d-appendix">
</div>


<!--radix_placeholder_site_after_body-->
<!--/radix_placeholder_site_after_body-->
<!--radix_placeholder_appendices-->
<div class="appendix-bottom">
<h3 id="updates-and-corrections">Corrections</h3>
<p>If you see mistakes or want to suggest changes, please <a href="https://github.com/bocasbiome/web/issues/new">create an issue</a> on the source repository.</p>
<h3 id="reuse">Reuse</h3>
<p>Text and figures are licensed under Creative Commons Attribution <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">CC BY 4.0</a>. Source code is available at <a href="https://github.com/bocasbiome/web/">https://github.com/bocasbiome/web/</a>, unless otherwise noted. The figures that have been reused from other sources don't fall under this license and can be recognized by a note in their caption: "Figure from ...".</p>
</div>
<script id="distill-bibliography" type="text/bibtex">
@article{callahan2016dada2,
  title={DADA2: high-resolution sample inference from Illumina amplicon data},
  author={Callahan, Benjamin J and McMurdie, Paul J and Rosen, Michael J and Han, Andrew W and Johnson, Amy Jo A and Holmes, Susan P},
  journal={Nature Methods},
  volume={13},
  number={7},
  pages={581},
  year={2016},
  url = {https://doi.org/10.1038/nmeth.3869},
}

@article{martin2011cutadapt,
  title={Cutadapt removes adapter sequences from high-throughput sequencing reads},
  author={Martin, Marcel},
  journal={EMBnet. journal},
  volume={17},
  number={1},
  pages={10--12},
  year={2011},
  url = {https://doi.org/10.14806/ej.17.1.200}
}

@article{quast2012silva,
  title={The SILVA ribosomal RNA gene database project: improved data processing and web-based tools},
  author={Quast, Christian and Pruesse, Elmar and Yilmaz, Pelin and Gerken, Jan and Schweer, Timmy and Yarza, Pablo and Peplies, J{\"o}rg and Gl{\"o}ckner, Frank Oliver},
  journal={Nucleic Acids Research},
  volume={41},
  number={D1},
  pages={D590--D596},
  year={2012},
  url = {https://doi.org/10.1093/nar/gks1219}
}

@article{mcmurdie2013phyloseq,
  title={phyloseq: an R package for reproducible interactive analysis and graphics of microbiome census data},
  author={McMurdie, Paul J and Holmes, Susan},
  journal={PLoS One},
  volume={8},
  number={4},
  pages={e61217},
  year={2013},
  url = {https://doi.org/10.1371/journal.pone.0061217}
}

@misc{wong2011points,
  title={Points of view: Color blindness},
  author={Wong, Bang},
  journal={Nature Methods},
  volume={8},
  number={6},
  pages={441},
  year={2011},
  url = {https://doi.org/10.1038/nmeth.1618}
}

@misc{lahti2017microbiome,
  author = {Lahti, Leo and Shetty Sudarshan and others},
  title = {Tools for microbiome analysis in R. Version 1.9.97},
  year = {2017},
  url = {https://github.com/microbiome/microbiome},
}

@article{eren2015anvi,
  title={Anvi’o: an advanced analysis and visualization platform for ‘omics data},
  author={Eren, A Murat and Esen, {\"O}zcan C and Quince, Christopher and Vineis, Joseph H and Morrison, Hilary G and Sogin, Mitchell L and Delmont, Tom O},
  journal={PeerJ},
  volume={3},
  pages={e1319},
  year={2015},
  url = {https://doi.org/10.7717/peerj.1319}
}

@article{bolger2014trimmomatic,
  title={Trimmomatic: a flexible trimmer for Illumina sequence data},
  author={Bolger, Anthony M and Lohse, Marc and Usadel, Bjoern},
  journal={Bioinformatics},
  volume={30},
  number={15},
  pages={2114--2120},
  year={2014},
  url = {https://doi.org/10.1093/bioinformatics/btu170}
}

@article{koster2012snakemake,
  title={Snakemake—a scalable bioinformatics workflow engine},
  author={K{\"o}ster, Johannes and Rahmann, Sven},
  journal={Bioinformatics},
  volume={28},
  number={19},
  pages={2520--2522},
  year={2012},
  url = {https://doi.org/10.1093/bioinformatics/bts480}
}

@article{menzel2016fast,
  title={Fast and sensitive taxonomic classification for metagenomics with Kaiju},
  author={Menzel, Peter and Ng, Kim Lee and Krogh, Anders},
  journal={Nature Communications},
  volume={7},
  pages={11257},
  year={2016},
  url = {https://doi.org/10.1038/ncomms11257}
}

@article{roux2015virsorter,
  title={VirSorter: mining viral signal from microbial genomic data},
  author={Roux, Simon and Enault, Francois and Hurwitz, Bonnie L and Sullivan, Matthew B},
  journal={PeerJ},
  volume={3},
  pages={e985},
  year={2015},
  url = {https://doi.org/10.1093/bioinformatics/btu170}
}

@article{kim2016centrifuge,
  title={Centrifuge: rapid and sensitive classification of metagenomic sequences},
  author={Kim, Daehwan and Song, Li and Breitwieser, Florian P and Salzberg, Steven L},
  journal={Genome Research},
  volume={26},
  number={12},
  pages={1721--1729},
  year={2016},
  url = {https://doi.org/10.1101/gr.210641.116.}
}

@article{hyatt2010prodigal,
  title={Prodigal: prokaryotic gene recognition and translation initiation site identification},
  author={Hyatt, Doug and Chen, Gwo-Liang and LoCascio, Philip F and Land, Miriam L and Larimer, Frank W and Hauser, Loren J},
  journal={BMC Bioinformatics},
  volume={11},
  number={1},
  pages={119},
  year={2010},
  url = {https://doi.org/10.1186/1471-2105-11-119}
}

@article{edgar2004muscle,
  title={MUSCLE: multiple sequence alignment with high accuracy and high throughput},
  author={Edgar, Robert C},
  journal={Nucleic Acids Research},
  volume={32},
  number={5},
  pages={1792--1797},
  year={2004},
  url = {https://doi.org/10.1093/nar/gkh340}
}

@article{li2015megahit,
  title={MEGAHIT: an ultra-fast single-node solution for large and complex metagenomics assembly via succinct de Bruijn graph},
  author={Li, Dinghua and Liu, Chi-Man and Luo, Ruibang and Sadakane, Kunihiko and Lam, Tak-Wah},
  journal={Bioinformatics},
  volume={31},
  number={10},
  pages={1674--1676},
  year={2015},
  url = {https://doi.org/10.1093/bioinformatics/btv033}
}

@article{langmead2012fast,
  title={Fast gapped-read alignment with Bowtie 2},
  author={Langmead, Ben and Salzberg, Steven L},
  journal={Nature Methods},
  volume={9},
  number={4},
  pages={357},
  year={2012},
  url = {https://doi.org/10.1038/nmeth.1923}
}

@article{li2009fast,
  title={Fast and accurate short read alignment with Burrows--Wheeler transform},
  author={Li, Heng and Durbin, Richard},
  journal={Bioinformatics},
  volume={25},
  number={14},
  pages={1754--1760},
  year={2009},
  url = {https://doi.org/10.1093/bioinformatics/btp324}
}

@article{li2009sequence,
  title={The sequence alignment/map format and SAMtools},
  author={Li, Heng and Handsaker, Bob and Wysoker, Alec and Fennell, Tim and Ruan, Jue and Homer, Nils and Marth, Gabor and Abecasis, Goncalo and Durbin, Richard},
  journal={Bioinformatics},
  volume={25},
  number={16},
  pages={2078--2079},
  year={2009},
  url = {https://doi.org/10.1093/bioinformatics/btp352}
}

@article{breitwieser2018krakenuniq,
  title={KrakenUniq: confident and fast metagenomics classification using unique k-mer counts},
  author={Breitwieser, FP and Baker, DN and Salzberg, Steven L},
  journal={Genome Biology},
  volume={19},
  number={1},
  pages={1--10},
  year={2018},
  url = {https://doi.org/10.5281/zenodo.1412647}
}

@article{eren2013filtering,
  title={A filtering method to generate high quality short reads using Illumina paired-end technology},
  author={Eren, A Murat and Vineis, Joseph H and Morrison, Hilary G and Sogin, Mitchell L},
  journal={PLoS One},
  volume={8},
  number={6},
  year={2013},
  url = {https://doi.org/10.1371/journal.pone.0066643}
}

@article{alneberg2014binning,
  title={Binning metagenomic contigs by coverage and composition},
  author={Alneberg, Johannes and Bjarnason, Brynjar Sm{\'a}ri and De Bruijn, Ino and Schirmer, Melanie and Quick, Joshua and Ijaz, Umer Z and Lahti, Leo and Loman, Nicholas J and Andersson, Anders F and Quince, Christopher},
  journal={Nature Methods},
  volume={11},
  number={11},
  pages={1144--1146},
  year={2014},
  url = {https://doi.org/10.1038/nmeth.3103}
}

@article{sieber2018recovery,
  title={Recovery of genomes from metagenomes via a dereplication, aggregation and scoring strategy},
  author={Sieber, Christian MK and Probst, Alexander J and Sharrar, Allison and Thomas, Brian C and Hess, Matthias and Tringe, Susannah G and Banfield, Jillian F},
  journal={Nature Microbiology},
  volume={3},
  number={7},
  pages={836--843},
  year={2018},
}

@article{kang2019metabat,
  title={MetaBAT 2: an adaptive binning algorithm for robust and efficient genome reconstruction from metagenome assemblies},
  author={Kang, Dongwan D and Li, Feng and Kirton, Edward and Thomas, Ashleigh and Egan, Rob and An, Hong and Wang, Zhong},
  journal={PeerJ},
  volume={7},
  pages={e7359},
  year={2019},
}
@article{wu2016maxbin,
  title={MaxBin 2.0: an automated binning algorithm to recover genomes from multiple metagenomic datasets},
  author={Wu, Yu-Wei and Simmons, Blake A and Singer, Steven W},
  journal={Bioinformatics},
  volume={32},
  number={4},
  pages={605--607},
  year={2016},
}

@article{ondov2011krona,
  title={Interactive metagenomic visualization in a Web browser},
  author={Ondov, Brian D and Bergman, Nicholas H and Phillippy, Adam M},
  journal={BMC Bioinformatics},
  volume={12},
  number={1},
  pages={385},
  year={2011},
  url = {https://doi.org/10.1186/1471-2105-12-385}
}

@article{buchfink2015fast,
  title={Fast and sensitive protein alignment using DIAMOND},
  author={Buchfink, Benjamin and Xie, Chao and Huson, Daniel H},
  journal={Nature Methods},
  volume={12},
  number={1},
  pages={59--60},
  year={2015},
  url = {https://doi.org/10.1038/nmeth.3176}
}

@article{delmont2018nitrogen,
  title={Nitrogen-fixing populations of Planctomycetes and Proteobacteria are abundant in surface ocean metagenomes},
  author={Delmont, Tom O and Quince, Christopher and Shaiber, Alon and Esen, {\"O}zcan C and Lee, Sonny TM and Rapp{\'e}, Michael S and McLellan, Sandra L and L{\"u}cker, Sebastian and Eren, A Murat},
  journal={Nature Microbiology},
  volume={3},
  number={7},
  pages={804--813},
  year={2018},
  url = {https://doi.org/10.1038/s41564-018-0176-9}
}

@article{dongen2000cluster,
  title={A cluster algorithm for graphs},
  author={Dongen, Stijn},
  year={2000},
  publisher={CWI (Centre for Mathematics and Computer Science)},
  url = {https://www.micans.org/mcl/index.html?sec_software}
}

@article{jain2018high,
  title={High throughput ANI analysis of 90K prokaryotic genomes reveals clear species boundaries},
  author={Jain, Chirag and Rodriguez-R, Luis M and Phillippy, Adam M and Konstantinidis, Konstantinos T and Aluru, Srinivas},
  journal={Nature Communications},
  volume={9},
  number={1},
  pages={1--8},
  year={2018},
  url={https://doi.org/10.1038/s41467-018-07641-9}
}

@article{stamatakis2014raxml,
  title={RAxML version 8: a tool for phylogenetic analysis and post-analysis of large phylogenies},
  author={Stamatakis, Alexandros},
  journal={Bioinformatics},
  volume={30},
  number={9},
  pages={1312--1313},
  year={2014},
  url={https://doi.org/10.1093/bioinformatics/btu033}
}

@article{minh2020iq,
  title={IQ-TREE 2: New models and efficient methods for phylogenetic inference in the genomic era},
  author={Minh, Bui Quang and Schmidt, Heiko A and Chernomor, Olga and Schrempf, Dominik and Woodhams, Michael D and Von Haeseler, Arndt and Lanfear, Robert},
  journal={Molecular Biology and Evolution},
  volume={37},
  number={5},
  pages={1530--1534},
  year={2020},
  url={https://doi.org/10.1093/molbev/msaa015}
}

@article{kalyaanamoorthy2017modelfinder,
  title={ModelFinder: fast model selection for accurate phylogenetic estimates},
  author={Kalyaanamoorthy, Subha and Minh, Bui Quang and Wong, Thomas KF and von Haeseler, Arndt and Jermiin, Lars S},
  journal={Nature Methods},
  volume={14},
  number={6},
  pages={587},
  year={2017},
  url={https://doi.org/10.1038/nmeth.4285}
}

@article{quince2011removing,
  title={Removing noise from pyrosequenced amplicons},
  author={Quince, Christopher and Lanzen, Anders and Davenport, Russell J and Turnbaugh, Peter J},
  journal={BMC bioinformatics},
  volume={12},
  number={1},
  pages={38},
  year={2011},
  url={https://doi.org/10.1186/1471-2105-12-38}
}
@article{caporaso2011global,
  title={Global patterns of 16S rRNA diversity at a depth of millions of sequences per sample},
  author={Caporaso, J Gregory and Lauber, Christian L and Walters, William A and Berg-Lyons, Donna and Lozupone, Catherine A and Turnbaugh, Peter J and Fierer, Noah and Knight, Rob},
  journal={Proceedings of the National Academy of Sciences},
  volume={108},
  number={Supplement 1},
  pages={4516--4522},
  year={2011},
  url={https://doi.org/10.1073/pnas.1000080107}
}
@article{kanehisa2000kegg,
  title={KEGG: kyoto encyclopedia of genes and genomes},
  author={Kanehisa, Minoru and Goto, Susumu},
  journal={Nucleic Acids Research},
  volume={28},
  number={1},
  pages={27--30},
  year={2000},
  url={https://doi.org/10.1093/nar/28.1.27}
}

@article{kanehisa2016blastkoala,
  title={BlastKOALA and GhostKOALA: KEGG tools for functional characterization of genome and metagenome sequences},
  author={Kanehisa, Minoru and Sato, Yoko and Morishima, Kanae},
  journal={Journal of Molecular Biology},
  volume={428},
  number={4},
  pages={726--731},
  year={2016},
  url={https://doi.org/10.1016/j.jmb.2015.11.006}
}

@article{minoche2011evaluation,
  title={Evaluation of genomic high-throughput sequencing data generated on Illumina HiSeq and genome analyzer systems},
  author={Minoche, Andr{\'e} E and Dohm, Juliane C and Himmelbauer, Heinz},
  journal={Genome Biology},
  volume={12},
  number={11},
  pages={R112},
  year={2011},
  url={https://doi.org/10.1186/gb-2011-12-11-r112}
}

@article{lee2019gtotree,
  title={GToTree: a user-friendly workflow for phylogenomics},
  author={Lee, Michael D},
  journal={Bioinformatics},
  volume={35},
  number={20},
  pages={4162--4164},
  year={2019},
  url={https://doi.org/10.1093/bioinformatics/btz188}
}

@article{wang2007naive,
  title={Naive Bayesian classifier for rapid assignment of rRNA sequences into the new bacterial taxonomy},
  author={Wang, Qiong and Garrity, George M and Tiedje, James M and Cole, James R},
  journal={Applied & Environmental Microbiology},
  volume={73},
  number={16},
  pages={5261--5267},
  year={2007},
  url={https://doi.org/10.1128/AEM.00062-07}
}

@article{hoang2017ultrafast,
  author = {Hoang, Diep Thi and Chernomor, Olga and von Haeseler, Arndt and Minh, Bui Quang and Vinh, Le Sy},
  title = {UFBoot2: Improving the Ultrafast Bootstrap Approximation},
  journal = {Molecular Biology and Evolution},
  volume = {35},
  number = {2},
  pages = {518-522},
  year = {2017},
  url = {https://doi.org/10.1093/molbev/msx281}
}

@article{parada2016every,
  title={Every base matters: assessing small subunit rRNA primers for marine microbiomes with mock communities, time series and global field samples},
  author={Parada, Alma E and Needham, David M and Fuhrman, Jed A},
  journal={Environmental microbiology},
  volume={18},
  number={5},
  pages={1403--1414},
  year={2016},
  url = {https://doi.org/10.1111/1462-2920.13023}
}

@article{parks2018gtgb,
  title={A standardized bacterial taxonomy based on genome phylogeny substantially revises the tree of life},
  author={Parks, Donovan H and Chuvochina, Maria and Waite, David W and Rinke, Christian and Skarshewski, Adam and Chaumeil, Pierre-Alain and Hugenholtz, Philip},
  journal={Nature Biotechnology},
  volume={36},
  number={10},
  pages={1546-1696},
  year={2018},
  url = {https://doi.org/10.1038/nbt.4229}
}

@article{foster2017metacoder,
  title={Metacoder: an R package for visualization and manipulation of community taxonomic diversity data},
  author={Foster, Zachary SL and Sharpton, Thomas J and Gr{\"u}nwald, Niklaus J},
  journal={PLoS Computational Biology},
  volume={13},
  number={2},
  pages={e1005404},
  year={2017},
  url={https://doi.org/10.1371/journal.pcbi.1005404}
}

@article{dufrene1997species,
  title={Species assemblages and indicator species: the need for a flexible asymmetrical approach},
  author={Dufr{\^e}ne, Marc and Legendre, Pierre},
  journal={Ecological Monographs},
  volume={67},
  number={3},
  pages={345--366},
  year={1997},
  url={https://doi.org/10.1890/0012-9615(1997)067[0345:SAAIST]2.0.CO;2}
}

@article{roberts2007labdsv,
  title={labdsv: Ordination and multivariate analysis for ecology},
  author={Roberts, David W and others},
  journal={R package version},
  volume={1},
  number={1},
  year={2007}
}
</script>
<!--/radix_placeholder_appendices-->
<!--radix_placeholder_navigation_after_body-->
<!--/radix_placeholder_navigation_after_body-->

</body>

</html>
